src/main/java/io/netty/incubator/codec/quic/QuicChannel.java

```diff
@@ -70,4 +70,9 @@
      * @return                  the future that is notified.
      */
     ChannelFuture close(boolean applicationClose, int error, ByteBuf reason, ChannelPromise promise);
+
+    /**
+     * Collects and returns statistics about the connection.
+     */
+    QuicConnectionStats stats();
```

Comment:
Generally speaking this seems to be fine we just need to be sure this is thread safe which means we need to ensure that this can't race with closing the underlying connection

Reply:
Yep as I've mentioned because of lifetimes mismatch there are a few corner cases here. I'm going to offload the operation onto `eventLoop` (if not there), I assume this would do a trick?

Reply:
Yes ... maybe rename it to collectStats or something as well

Reply:
The one situation where `collectStats` name would make less sense is when you ask for stats from a channel after the connection was already dropped. Simplest way is to implement `channelInactive`. Or even in general, I can hold pointer to a channel object for as long as I want. If I try to call `collectStats`, it cannot deref implementation to a native layer as the native pointer is already gone. So, what I'm doing is keeping `statsAtClose` object  that I populate before calling `quiche_conn_free`. In this case function would not "collect" stats, it would return the latest stats that were available.

Reply:
But wouldn't you may want to collect these before close as well ?

Reply:
In a way yes.. I will rename the method!